src/App.tsx

```diff
@@ -1,26 +1,355 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import {
+  createTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+  USER_ID,
+} from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import classNames from 'classnames';
+import { Footer } from './components/Footer';
+import { Filter } from './utils/Enums';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>(todos);
+  const [title, setTitle] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<Set<number>>(new Set());
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filter, setFilter] = useState(Filter.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [titleEditingId, setTitleEditingId] = useState<null | number>(null);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const completedTodos = todos.filter(todo => todo.completed);
+  const notCompletedTodos = todos.filter(todo => !todo.completed);
+
+  // eslint-disable-next-line @typescript-eslint/no-shadow
+  function filterTodos(todos: Todo[], filter: string) {
```

Comment:
You must follow one way of creating functions, fix it everywhere
```suggestion
  const filterTodos = (todos: Todo[], filter: string) => {
```